integration-commons/src/main/java/org/symphonyoss/integration/logging/LogMessageSource.java

```diff
@@ -0,0 +1,102 @@
+package org.symphonyoss.integration.logging;
+
+import com.symphony.logging.ISymphonyLogger;
+import com.symphony.logging.SymphonyLoggerFactory;
+
+import org.apache.commons.io.FilenameUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.NoSuchMessageException;
+import org.springframework.context.support.MessageSourceAccessor;
+import org.springframework.context.support.ResourceBundleMessageSource;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.core.io.support.ResourcePatternUtils;
+import org.springframework.stereotype.Component;
+import org.symphonyoss.integration.exception.ExceptionMessageFormatter;
+
+import java.io.IOException;
+import java.util.Locale;
+
+import javax.annotation.PostConstruct;
+
+/**
+ * This utility class provides access and formatting methods for message resources from resource bundle files, located
+ * at all of the application class paths. The resource bundles should contain application logging messages only,
+ * as the purpose of this component is supporting the construction of output information for application logging.
+ *
+ * In order to use this component, add a your-component-log-messages.properties to your component's classpath. The file
+ * will be automatically loaded as long as it ends with "-log-messages.properties". Make sure you create a unique name
+ * for your file, so that your resources don't get overridden by another conflicting file. Likewise, make sure the
+ * resource ids on your file have a name convention that avoid conflicts with other resources, i.e. avoid simple names
+ * such as "configuration.not.found". Add your component identifier as a prefix for resource ids to avoid such
+ * problem.
+ *
+ * Resources from all application components are made available through this component, but different application
+ * components should not rely on each other resources, unless they are naturally dependent on each other.
+ *
+ * Created by Evandro Carrenho on 30/01/17.
+ */
+@Component
+public class LogMessageSource {
+
+  private static final ISymphonyLogger LOGGER = SymphonyLoggerFactory.getLogger(LogMessageSource.class);
+
+  private static final String UNABLE_TO_LOAD_LOG_MESSAGES = "integration.commons.unable.to.load.log.messages";
+
+  private static final String UNABLE_TO_LOAD_LOG_MESSAGES_SOLUTION1 =
```

Comment:
Will remove it, I was numbering solutions as they might have been multiple, but it ended up being weird.